### PR TITLE
Support stackstrace for aarch64

### PR DESCRIPTION
**Pull request by huangtianhua**

### CODE REVIEW

**Comment by huangtianhua** [comment]
@googlebot I signed it!

File: src/stacktrace_aarch64-inl.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+//
+// Author: Craig Silverstein
+//
+// Produce stack trace.  I'm guessing (hoping!) the code is much like
+// for x86.  For apple machines, at least, it seems to be; see
+//    http://developer.apple.com/documentation/mac/runtimehtml/RTArch-59.html
+//    http://www.linux-foundation.org/spec/ELF/ppc64/PPC-elf64abi-1.9.html#STACK
+// Linux has similar code: http://patchwork.ozlabs.org/linuxppc/patch?id=8882
+
+#include <stdio.h>
```
Comment (sergiud) [inline src/stacktrace_aarch64-inl.h]: Please use `<cstdio>`.

File: src/stacktrace_aarch64-inl.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#include <stdint.h>   // for uintptr_t
```
Comment (sergiud) [inline src/stacktrace_aarch64-inl.h]: One should include `<cstdint>`. However, the header is available only starting with C++11. Glog still uses C++03.

File: bazel/glog.bzl [hunk 3]
```diff
@@ -83,6 +83,7 @@ def glog_library(namespace = "google", with_gflags = 1, **kwargs):
             "src/stacktrace_generic-inl.h",
             "src/stacktrace_libunwind-inl.h",
             "src/stacktrace_powerpc-inl.h",
+	    "src/stacktrace_aarch64-inl.h",
```
Comment (sergiud) [inline bazel/glog.bzl]: Nitpick: inconsistent indentation.

Comment (huangtianhua) [reply bazel/glog.bzl]: Done.

File: src/stacktrace_aarch64-inl.h (hunk 1 above)
Comment (huangtianhua) [reply src/stacktrace_aarch64-inl.h]: I find other files use <stdio.h>?

File: src/stacktrace_aarch64-inl.h (hunk 2 above)
Comment (huangtianhua) [reply src/stacktrace_aarch64-inl.h]: same

**Comment by huangtianhua** [comment]
@sergiud , Thanks for reviewing. And I have a question about getting stacktrace by libunwind, if HAVE_LIB_UNWIND is set(like the case apache/kudu using), we will get stack trace by libunwind before my commit, but seems the function 'GetStackTrace' of stacktrace_libunwind-inl.h dosen't work on aarch64, so I changed the logic to use stacktrace_aarch64-inl.h(which I newly added) if HAVE_LIB_UNWIND is set. So I wonder it is the right way to get stack trace by libunwind on aarch64?  or it doesn't support on aarch64? or how to modify the file stacktrace_libunwind-inl.h to support get stack trace on aarch64? Thanks very much.

File: src/stacktrace_aarch64-inl.h (hunk 1 above)
Comment (sergiud) [reply src/stacktrace_aarch64-inl.h]: Ok then.

File: src/stacktrace_aarch64-inl.h (hunk 2 above)
Comment (sergiud) [reply src/stacktrace_aarch64-inl.h]: Not quite. You need a fallback to C++03 where the header is not available.

Comment (huangtianhua) [reply src/stacktrace_aarch64-inl.h]: Other files include <stdint.h>, so I think it works.

Comment (sergiud) [reply src/stacktrace_aarch64-inl.h]: Including is okay, but you still need a fallback.

**Comment by romange** [comment]
@huangtianhua would it be possible to finish this PR?

**Comment by sergiud** [comment]
@romange Could you please test the changes?

**Review by sergiud — APPROVED** [review]

**Comment by romange** [comment]
@sergiud  The bug in libunwind seems to be fixed https://github.com/libunwind/libunwind/issues/235  

Do you still want to commit the workaround for aarch64 or you prefer to wait for the next version of libunwind to be released?

**Comment by sergiud** [comment]
@romange Thanks for the info. I'd rather avoid the workaround.

**Pull request closed without merging**